benchmarks/common/benchmarks/toy_detector_benchmark.hpp

```diff
@@ -0,0 +1,184 @@
+    static const int n_tracks = 5000u;
+
+    std::vector<traccc::spacepoint_collection_types::host> spacepoints;
+    std::vector<traccc::measurement_collection_types::host> measurements;
+
+    // Configs
+    traccc::seedfinder_config seeding_cfg;
+    traccc::seedfilter_config filter_cfg;
+    traccc::spacepoint_grid_config grid_cfg{seeding_cfg};
+    traccc::finding_config<float> finding_cfg;
+    traccc::fitting_config fitting_cfg;
+
+    static constexpr std::array<float, 2> phi_range{
+        -traccc::constant<float>::pi, traccc::constant<float>::pi};
+    static constexpr std::array<float, 2> theta_range{
+        0.f, traccc::constant<float>::pi};
+    static constexpr std::array<float, 2> mom_range{
+        10.f * traccc::unit<float>::GeV, 100.f * traccc::unit<float>::GeV};
+
+    static inline const std::string sim_dir = "toy_detector_benchmark/";
+
+    // Detector type
+    using detector_type = detray::detector<detray::toy_metadata>;
+
+    // B field value and its type
+    // @TODO: Set B field as argument
+    using b_field_t = covfie::field<detray::bfield::const_bknd_t>;
+
+    static constexpr traccc::vector3 B{0, 0,
+                                       2 * detray::unit<traccc::scalar>::T};
+
+    ToyDetectorBenchmark() {
+
+        std::cout << "Please be patient. It may take some time to generate "
+                     "the simulation data."
+                  << std::endl;
+
+        // VecMem memory resource(s)
+        vecmem::host_memory_resource host_mr;
+
+        // Use deterministic random number generator for testing
+        using uniform_gen_t = detray::detail::random_numbers<
+            traccc::scalar, std::uniform_real_distribution<traccc::scalar>>;
+
+        // Build the detector
+        auto [det, name_map] =
+            detray::build_toy_detector(host_mr, get_toy_config());
+
+        // B field
+        auto field = detray::bfield::create_const_field(B);
+
+        // Origin of particles
+        using generator_type =
+            detray::random_track_generator<traccc::free_track_parameters,
+                                           uniform_gen_t>;
+        generator_type::configuration gen_cfg{};
+        gen_cfg.n_tracks(n_tracks);
+        gen_cfg.phi_range(phi_range);
+        gen_cfg.theta_range(theta_range);
+        gen_cfg.mom_range(mom_range);
+        generator_type generator(gen_cfg);
+
+        // Smearing value for measurements
+        traccc::measurement_smearer<traccc::default_algebra> meas_smearer(
+            50 * detray::unit<traccc::scalar>::um,
+            50 * detray::unit<traccc::scalar>::um);
+
+        // Type declarations
+        using writer_type = traccc::smearing_writer<
+            traccc::measurement_smearer<traccc::default_algebra>>;
+
+        // Writer config
+        typename writer_type::config smearer_writer_cfg{meas_smearer};
+
+        // Run simulator
+        const std::string full_path = traccc::io::data_directory() + sim_dir;
```

Comment:
We will really need to find a better procedure for using temporary directories. :thinking:

But since that will be necessary in other places as well, we can do that in a future PR...